src/containers/WarningConfirmThresholdScenario/index.jsx

```diff
@@ -49,28 +49,50 @@ class WarningConfirmThresholdScenario extends React.Component {
 		if (!isValid) {
 			return;
 		}
-		const nextTreshold = this.props.treshold.value;
-		let { permissionsKeys } = this.props;
-		const { network, account } = this.props;
-		const accountId = account.toJS().id;
-		permissionsKeys = permissionsKeys.toJS();
-		const activeKeysData = {
-			keys: permissionsKeys.active.keys.concat(permissionsKeys.active.accounts),
-			threshold: permissionsKeys.active.threshold,
-		};
-		const maxNextThreshold = activeKeysData.keys
-			.reduce((accumulator, currentKey) => accumulator + currentKey.weight, 0);
-		if (maxNextThreshold < nextTreshold) {
+		const permissionsKeys = this.props.form.toJS();
+		const network = this.props.network.toJS();
+		const accs = JSON.parse(localStorage.getItem(`accounts_${network.name}`));
+		const nextTreshold = new BN(this.props.treshold.value);
+		const { keys, accounts } = permissionsKeys.active;
+		let enoughNextThreshold = new BN(0);
+		let maxNextValue = new BN(0);
+		for (const key in keys) {
+			maxNextValue = maxNextValue.plus(keys[key].weight.value);
+			if (keys[key].hasWif && keys[key].hasWif.value) {
+				enoughNextThreshold = enoughNextThreshold.plus(keys[key].weight.value);
+			}
+		}
+		for (const account in accounts) {
+			maxNextValue = maxNextValue.plus(accounts[account].weight.value);
+			for (let i = 0; i < accs.length; i += 1) {
+				if (accs[i].name === account) {
+					enoughNextThreshold =
+						enoughNextThreshold.plus(accounts[account].weight.value);
+				}
+			}
+		}
+		const { echoRand } = permissionsKeys;
+		for (const key in echoRand.keys) {
+			if (!echoRand.keys[key].hasWif.value) {
+				this.setState({
+					echoRandMessage:
+						'You remove your EchoRandKey and now will lose access to it.',
```

Comment:
replace to new text
`You removed your EchoRand Key and after submitting this action you will lose access to it.`